conjurapi/authn/oidc_authenticator.go

```diff
@@ -0,0 +1,31 @@
+package authn
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/cyberark/conjur-api-go/conjurapi/response"
+)
+
+type OIDCAuthenticator struct {
+	AuthenticateURL string `env:"CONJUR_AUTHN_OIDC_URL"`
+	IDToken         string `env:"CONJUR_AUTHN_OIDC_ID_TOKEN"`
+}
+
+func (a *OIDCAuthenticator) RefreshToken() ([]byte, error) {
+	v := url.Values{}
+	v.Set("id_token", a.IDToken)
+
+	resp, err := http.PostForm(a.AuthenticateURL, v)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return response.DataResponse(resp)
+}
+
+func (a *OIDCAuthenticator) NeedsTokenRefresh() bool {
+	// We're not going to implement a refresh token flow...
+	return false
```

Comment:
Wouldn't this mean that the conjur token is invalid after 8 mins and won't be able to pull secrets anymore for the app that uses this? If so, maybe it would be good to see if we can reuse some parts of other authenticators here that refresh this token.